src/Database/Bloodhound/Types.hs

```diff
@@ -4373,6 +4667,356 @@ instance FromJSON NodesInfo where
+
+instance FromJSON NodeProcessStats where
+  parseJSON = withObject "NodeProcessStats" parse
+    where
+      parse o = do
+        mem <- o .: "mem"
+        cpu <- o .: "cpu"
+        NodeProcessStats <$> mem .: "total_virtual_in_bytes"
+                         <*> mem .: "share_in_bytes"
+                         <*> mem .: "resident_in_bytes"
+                         <*> (unMS <$> cpu .: "total_in_millis")
+                         <*> (unMS <$> cpu .: "user_in_millis")
+                         <*> (unMS <$> cpu .: "sys_in_millis")
+                         <*> cpu .: "percent"
+                         <*> o .: "open_file_descriptors"
+                         <*> (posixMS <$> o .: "timestamp")
+
+instance FromJSON NodeOSStats where
+  parseJSON = withObject "NodeOSStats" parse
+    where
+      parse o = do
+        swap <- o .: "swap"
+        mem <- o .: "mem"
+        cpu <- o .: "cpu"
+        load <- o .: "load_average"
+        NodeOSStats <$> swap .: "free_in_bytes"
+                    <*> swap .: "used_in_bytes"
+                    <*> mem .: "actual_used_in_bytes"
+                    <*> mem .: "actual_free_in_bytes"
+                    <*> mem .: "used_percent"
+                    <*> mem .: "free_percent"
+                    <*> mem .: "used_in_bytes"
+                    <*> mem .: "free_in_bytes"
+                    <*> cpu .: "stolen"
+                    <*> cpu .: "usage"
+                    <*> cpu .: "idle"
+                    <*> cpu .: "user"
+                    <*> cpu .: "sys"
+                    <*> pure load
+                    <*> (unMS <$> o .: "uptime_in_millis")
+                    <*> (posixMS <$> o .: "timestamp")
+
+instance FromJSON LoadAvgs where
+  parseJSON = withArray "LoadAvgs" parse
+    where
+      parse v = case V.toList v of
+        [one, five, fifteen] -> LoadAvgs <$> parseJSON one
+                                         <*> parseJSON five
+                                         <*> parseJSON fifteen
+        _                    -> fail "Expecting a triple of Doubles"
+
+instance FromJSON NodeIndicesStats where
+  parseJSON = withObject "NodeIndicesStats" parse
+    where
+      parse o = do
+        recovery <- o .: "recovery"
+        queryCache <- o .: "query_cache"
+        suggest <- o .: "suggest"
+        translog <- o .: "translog"
+        segments <- o .: "segments"
+        completion <- o .: "completion"
+        percolate <- o .: "percolate"
+        fielddata <- o .: "fielddata"
+        idCache <- o .: "id_cache"
+        filterCache <- o .: "filter_cache"
+        warmer <- o .: "warmer"
+        flush <- o .: "flush"
+        refresh <- o .: "refresh"
+        merges <- o .: "merges"
+        search <- o .: "search"
+        getStats <- o .: "get"
+        indexing <- o .: "indexing"
+        store <- o .: "store"
+        docs <- o .: "docs"
+        NodeIndicesStats <$> (unMS <$> recovery .: "throttle_time_in_millis")
```

Comment:
Yeah. And it gets hairier since I'm finding from these test failures that the higher level keys like "recovery" are optional, so I'm defining a little scoped combinator `(.::)` That works where the object is a Maybe. You know how folks get about lenses though.